metadata/metadata.go

```diff
@@ -47,6 +48,16 @@ type MDSClientInterface interface {
 	WriteGuestAttributes(context.Context, string, string) error
 }
 
+// requestConfig is used internally to configure an http request given its context.
+type requestConfig struct {
```

Comment:
If we want to move all HTTP calls to MDS to a central location I think headers and body should also be included in requestConfig or you're thinking of keeping write calls separate?

Reply:
For now I don't plan to add write calls support to `requestConfig` or `client.do()`, at least not with this PR.

Reply:
for that purpose, I'm even dropping the `method` field.

Reply:
Done. Thanks.